src/dune_rules/context.ml

```diff
@@ -80,33 +80,52 @@ module Env_nodes = struct
   ;;
 end
 
-type t =
-  { name : Context_name.t
-  ; kind : Kind.t
-  ; profile : Profile.t
+type shared =
```

Comment:
So this is just currying some shared arguments in the later functions?

Reply:
Yes, it just saves us passing a bunch of arguments in a couple of places.